oauth2_http/javatests/com/google/auth/oauth2/OAuth2CredentialsTest.java

```diff
@@ -873,6 +876,83 @@ public void serialize() throws IOException, ClassNotFoundException {
     assertSame(deserializedCredentials.clock, Clock.SYSTEM);
   }
 
+  @Test
+  public void UpdateTokenValueBeforeWake() throws IOException, InterruptedException {
```

Comment:
Nit: methods should always use lowerCamelCase

Reply:
Also this test is very specific.

I wonder if it would make sense to add a smoke test that creates 500 creds and have 1000 threads race to refresh and get() the creds

Reply:
This test makes sure there is no regression of the race condition.

I think smoke tests could make sense but I do not like how flakey / hard to debug they can be. 

We could add one later, that doesn't run with the unit test suite